src/credentials/GroupDataProviderImpl.cpp

```diff
@@ -100,6 +108,59 @@ struct LinkedData : public PersistentData<kPersistentBufferMax>
     bool first      = true;
 };
 
+struct FabricList : public PersistentData<kPersistentBufferMax>
+{
+    static constexpr TLV::Tag TagFirstFabric() { return TLV::ContextTag(1); }
+    static constexpr TLV::Tag TagFabricCount() { return TLV::ContextTag(2); }
+
+    chip::FabricIndex first_fabric = kUndefinedFabricIndex;
+    uint16_t fabric_count          = 0;
```

Comment:
What is the rationale for the use of the uint16_t type for fabric_count?  It seems that for the most part first_fabric and fabric_count are treated as uint16 in this logic.  I assume though that because `FabricIndex` is uint8, that type would suffice.

Is there some subtle logic here that is accommodating duplicate fabric index entries and, therefore, a uint8 fabric_count isn't wide enough?

Reply:
Good observation, uint8_t shall suffice.